libs/clients/middlewares/src/lib/withErrorLog.ts

```diff
@@ -16,28 +18,51 @@ export function withErrorLog({
   logger,
 }: ErrorLogOptions): MiddlewareAPI {
   return (request) => {
-    return fetch(request).catch((error: Error) => {
-      const logLevel =
-        error instanceof FetchError &&
-        error.status < 500 &&
-        !treat400ResponsesAsErrors
-          ? 'warn'
-          : 'error'
-      const cacheStatus =
-        (error instanceof FetchError &&
-          error.response.headers.get('cache-status')) ??
-        undefined
+    islandis_logger.debug(
+      `Extended fetch request. ${JSON.stringify(
+        pick(request, [
+          'url',
+          'agent',
+          'json',
+          'headers',
+          'auth',
+          'method',
+          'timeout',
+          'body',
+        ]),
+      )}`,
+    )
```

Comment:
Open to this feature, but want a better implementation. 😇 Maybe in a new PR?

- Should use the `logger` instance passed to error log.
- Should check `logger.isDebugEnabled()` to avoid picking and stringifying request objects in production.
- Should not be in `withErrorLog.ts`. Can rename, or maybe better to keep separate and conditionally add `withLogging()` plugin using some option or logger.isDebugEnabled().
- Tests? :)

Reply:
Yup thats not supposed to be there 🤕 . Was a part of some debugging on this feature branch. I will remove.

Reply:
But I agree it would be nice to have some debugging options available here.